src/components/Radio/Radio.tsx

```diff
@@ -1,66 +1,157 @@
 import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+import Icon from '../Icon/Icon'
 
 export interface IRadioProps {
   as?: any
+  checked?: boolean
   children?: ReactChildren
   className?: string
+  disabled?: boolean
   label?: string
+  name?: string
+  onChange?: ComponentEventHandler<IRadioProps>
   type?: string
   styles?: IComponentPartStylesInput
+  value?: string | number
   variables?: ComponentVariablesInput
 }
 
 /**
  * @accessibility
- * This is shown at the top.
+ * Radio buttons do not handle accessibility. Those aspects will be handled once the RadioGroup component is introduced.
  */
-class Radio extends UIComponent<Extendable<IRadioProps>, any> {
+class Radio extends AutoControlledComponent<Extendable<IRadioProps>, any> {
+  static create: Function
+
   static displayName = 'Radio'
 
   static className = 'ui-radio'
 
   static propTypes = {
     as: customPropTypes.as,
 
+    /** Whether or not radio is checked. */
+    checked: PropTypes.bool,
+
     /** Child content * */
     children: PropTypes.node,
 
     /** Additional classes. */
     className: PropTypes.string,
 
+    /** Initial checked value. */
+    defaultChecked: PropTypes.bool,
+
+    /** A radio can appear disabled and be unable to change states. */
+    disabled: PropTypes.bool,
+
     /** The label of the radio input. */
     label: PropTypes.string,
 
+    /** The HTML input name. */
+    name: PropTypes.string,
+
+    /**
+     * Called after radio is clicked.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onChange: PropTypes.func,
+
     /** Custom styles to be applied for component. */
     styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
 
     /** The HTML input type. */
     type: PropTypes.string,
 
+    /** The HTML input value. */
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
     /** Custom variables to be applied for component. */
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
   }
 
-  static handledProps = ['as', 'children', 'className', 'label', 'styles', 'type', 'variables']
+  static handledProps = [
+    'as',
+    'checked',
+    'children',
+    'className',
+    'defaultChecked',
+    'disabled',
+    'label',
+    'name',
+    'onChange',
+    'styles',
+    'type',
+    'value',
+    'variables',
+  ]
 
   static defaultProps = {
     as: 'div',
     type: 'radio',
   }
 
-  renderComponent({ ElementType, classes, rest, styles }) {
-    const { type, label } = this.props
+  static autoControlledProps = ['checked']
+
+  private handleChange = (e: React.SyntheticEvent) => {
+    const { onChange, disabled } = this.props
+    const checked = (e.target as HTMLInputElement).checked
+
+    if (disabled) {
+      e.preventDefault()
+      return
+    }
+
+    this.trySetState({ checked })
+
+    if (onChange) {
```

Comment:
`_.invoke()` does the check for you

Reply:
I replaced this with _.invoke, but I prefer much this type of invoking because I know for sure when clicking the property that it is used and invoke somewhere in the code, and with using the invoke, the name of the callback is just a string.